src/transformStateWithClones.js

```diff
@@ -7,7 +7,44 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    const { type, keysToRemove, extraData } = action;
+    let newState;
+
+    switch (type) {
+      case 'addProperties':
+        newState = Object.assign(copyState, extraData);
+        break;
+
+      case 'removeProperties':
+        newState = { ...copyState };
+
+        for (const key of keysToRemove) {
+          delete newState[key];
+        }
+        break;
+
+      case 'clear':
+        newState = { ...copyState };
+
+        for (const key in newState) {
+          delete newState[key];
+        }
+        break;
```

Comment:
The task says that the 'clear' condition should create an empty object, so this code element can be simplified in this way:
`case 'clear':
    newState = {};
    break;`

Reply:
improve this moment))